src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,37 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+public class Contact {
+
+    public final String value;
+    private boolean isPrivate;
+  
+    /**
+     * Validates given contact.
+     *
+     * @throws IllegalValueException if given contact string is invalid.
+     */    
+    public Contact(String contact, boolean isPrivate) throws IllegalValueException {
+        this.isPrivate = isPrivate;
+        this.value = contact.trim();
+    }
+    
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof Contact // instanceof handles nulls
+                && this.value.equals(((Contact) other).value)); // state check
+    }
+    
+    public String toString() {
```

Comment:
I guess you should mark these methods are overridden methods.
From [Java documentation](https://docs.oracle.com/javase/tutorial/java/annotations/predefined.html):

> While it is not required to use this annotation when overriding a method, it helps to prevent errors. If a method marked with @Override fails to correctly override a method in one of its superclasses, the compiler generates an error.